src/qibo/tests/test_derivative.py

```diff
@@ -54,3 +56,39 @@ def test_derivative(backend):
     backend.assert_allclose(grad_0, 8.51104358e-02, atol=1e-10)
     backend.assert_allclose(grad_1, 5.20075970e-01, atol=1e-10)
     backend.assert_allclose(grad_2, 0, atol=1e-10)
+
+    # ------------------- rescaled parameter shift rule ------------------------
+
```

Comment:
I would suggest to split this to two test methods `test_parameter_shift` and `test_rescaled_parameter_shift`, instead of seperating with the comment. This way it will be easier to read and they will appear seperately in the pytest report, so it will be easier to identify issues in case one of them fails.

Reply:
Since I have removed the new method and modified the old one, this split is no longer necessary.